sitemap_utils.py

```
@@ -134,85 +135,89 @@ def _parse_sitemap_xml(self, content: str) -> Tuple[List[SitemapURL], List[str]]
-
         try:
-            # Handle XML namespaces
-            ns_match = re.search(r'xmlns\s*=\s*["\']([^"\']+)["\']', content)
-            namespace = ns_match[1] if ns_match else None
-
-            # Define namespace mapping
+            # Setup XML parsing with namespace support
+            namespace = self._extract_namespace(content)
             ns_map = {"sm": namespace} if namespace else {}
-
             root = ET.fromstring(content)
```

Ruff (0.8.2)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)